rust/agama-dbus-server/src/questions.rs

```diff
@@ -227,11 +244,43 @@ impl Questions {
         Ok(())
     }
 
-    /// sets questions to be answered by default answer instead of asking user
-    async fn use_default_answer(&mut self) -> Result<(), Error> {
-        log::info!("Answer questions with default option");
-        self.answer_strategies.push(Box::new(DefaultAnswers {}));
-        Ok(())
+    /// property that defines if questions is interactive or automatically answered with
+    /// default answer
+    #[dbus_interface(property)]
+    fn interactive(&self) -> bool {
+        let last = self.answer_strategies.last();
+        if let Some(real_strategy) = last {
+            real_strategy.id() != DefaultAnswers::id()
```

Comment:
I am not sure if I understood the idea of having more than one strategy. Is it for priorities? First default, if no found file, if not found ask to user? And the mode is only considered interactive if the last assigned strategy is not DefaultAnswers. But what happens if the last strategy is answers?

Reply:
Reason for strategies is extensibility, which allows to add completely different source of answers from strategy like e.g. some remote server answering it.
So whole idea is to have list of strategies and the first one that can answer it, provide that answer.
Default strategy should be always able to answer, so it is reason why it is last as answer file has precedence. And also it means that it is non interactive.
If last strategy is answers and it does not provide answer to given question, it is asked user. It is general rule, if no strategy can answer, then ask user ( aka interactive mode ).